appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AmountChooserPresenter.kt

```diff
@@ -29,6 +29,27 @@ class AmountChooserPresenter(private val appIconUrl: String, private val balance
         view?.sendKin(amount)
     }
 
+    override fun afterTextChanged(editable: Editable?) {
+        editable?.let {
+            if (editable.length >= 2) {
+                if (editable[0] == '0') {
+                    editable.delete(0, 1)
+                }
+            }
+            amount = 0
+            if (editable.isNotEmpty()) {
+                amount = try {
+                    editable.toString().toInt()
+                } catch (e: NumberFormatException) {
+                    editable.delete(editable.length - 2, editable.length - 1)
```

Comment:
what if the length is 0 or 1 ?

Reply:
in any case I am not sure I understand what you are trying to do here...

Reply:
i will write comment

Reply:
actually length 0 is not possible because you checked for notEmpty but couldn't it be that the user entered a single character that is not a number? In such a case editable.length-2 would be equal to -1.

Reply:
cant enter not number cause input type is number - keyboard only numbers